library/acl.inc

```diff
@@ -828,7 +828,9 @@
             return false;
         elseif($count_user_allowed > 0)
             return true;
-        elseif($count_group_denied > 0)
+        // We do not exclude user that belongs to group A and B
+        // In case of A group have permissions and B group does not have permissions
+        elseif($count_group_denied > 0 && $count_group_allowed < 1)
```

Comment:
O.k. I have removed this code and I will re-examine this if there is another insight to this I will commit it in a new branch